_plugins/sidebar.rb

```diff
@@ -0,0 +1,142 @@
+require "jekyll"
+require "nokogiri"
+require "json"
+
+module Jekyll
+  class Sidebar
+    attr_accessor :hash_pre_render
+    attr_accessor :filename_with_headers
+
+    def initialize
+      @hash_pre_render = {}
+      @filename_with_headers = {}
+    end
+
+    def pre_render(page)
+      menu_order = unless page["menu-order"].nil? then page["menu-order"] else 0 end
+      url = page["url"].gsub("index.html", "").gsub(".html", "")
+      @hash_pre_render[url] = {
+        :title => page["title"],
+        :url => page["url"],
+        :name => page["name"],
+        :menu_order => menu_order,
+      }
+    end
+
+    def post_write(site)
+      files = []
+      Dir.glob("#{site.config["destination"]}/**/*.html") do |filename|
+        doc = File.open(filename) { |f| Nokogiri::HTML(f) }
+        files.push(doc)
+        title = doc.title
+        url = ""
+        doc.xpath("/html/head/meta").each do |meta|
+          if meta.attr("property").eql? "og:url"
+            url = meta.attr("value")
+          end
+        end
+
+        headers = []
+        doc.xpath("//h1 | //h2 | //h2 | //h3 | //h4 | //h5 | //h6").each do |header|
+          if not header["id"]
+            next
+          end
+          child = header.last_element_child
+          header = {
+            :id => header["id"],
+            :title => header.content.strip,
+            :url => "#{url}#{child["href"]}",
+          }
+          headers.push(header)
+        end
+        sanitized_filename = filename.match(/(?m)(?<=\b_site).*$/)[0]
+        sanitized_filename = sanitized_filename.gsub("index.html", "")
+        sanitized_filename = sanitized_filename.gsub(".html", "")
+        @filename_with_headers[sanitized_filename] = { :headers => headers }
+      end
+
+      sidebar = render
+      files.each do | file |
+        file.xpath("//*[@id=\"dg-sidebar\"]").each do | location |
+          location.content = sidebar
+        end
+      end
+      #raise "undead"
+    end
+
+    def render
+      sidebar = "<div id=\"dg-sidebar\" class=\"sidebar\">"
+      sidebar << "<nav class=\"sidebar-nav\">"
+      sidebar << "<ul class=\"main-nav-ul\">"
+
+      merged = merge(@hash_pre_render, @filename_with_headers)
+      merged.each do |key, value|
```

Comment:
Codacy found an issue: [Unused block argument - `key`. If it's necessary, use `_` or `_key` as an argument name to indicate that it won't be used.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-design-guide-jekyll-theme/pullRequest?prid=5717102)